dali/base/dasds.cpp

```diff
@@ -3675,6 +3790,78 @@ class CLockInfo : public CInterface, implements IInterface
         return out;
     }
 
+    bool getFilteredLocks(CLockFilters &filters, MemoryBuffer &mb)
+    {
+        bool isFile;
+        StringBuffer file;
+        if (!filters.checkAndReadPath(xpath, isFile, file))
+            return false;
+
+        unsigned nlocks=0;
+        MemoryBuffer locks, localMB;
+        UInt64Array keys;
+        {
+            CHECKEDCRITICALBLOCK(crit, fakeCritTimeout);
+            HashIterator iter(connectionInfo);
+            ForEach(iter)
+            {
+                IMapping &imap = iter.query();
+                LockData *lD = connectionInfo.mapToValue(&imap);
+                keys.append(* ((ConnectionId *) imap.getKey()));
+                locks.append(sizeof(LockData), lD);
+                ++nlocks;
+            }
```

Comment:
This is duplicate of code in the pre-existing getLockInfo() .. should avoid code duplication like this.